src/Uno.UI/UI/Xaml/ResourceDictionary.cs

```diff
@@ -64,11 +64,19 @@ public bool HasKey(object key)
 			return _values.ContainsKey(keyName);
 		}
 
+		/// <summary>
+		/// Adds a new entry to the ResourceDictionary.
+		/// </summary>
+		/// <param name="key">The object key for the resource to insert.</param>
+		/// <param name="value">The object value for the resource to insert.</param>
+		/// <returns>True if calling the method replaced a value that already existed for the key;
+		/// false if calling the method defined a new key.</returns>
 		public bool Insert(object key, object value)
 		{
-			_values[key] = value;
-			return true;
-		}
+			var existed = _values.ContainsKey(key);
+			Set(key, value);
+			return existed;
```

Comment:
Could you add a test for that behavior as well ?

Reply:
Done